lib/iris/cube.py

```diff
@@ -1050,6 +1114,25 @@ def coord_dims(self, coord):
                        factories]
 
         if not matches:
+            raise iris.exceptions.CellMeasureNotFoundError(coord.name())
+
+        return matches[0]
+
+    def cell_measure_dims(self, cell_measure):
+        """
+        Returns a tuple of the data dimensions relevant to the given
+        CellMeasure.
+
+        * cell_measure
+            The CellMeasure to look for.
+
+        """
+        # Search for existing cell measure (object) on the cube, faster lookup
+        # than equality - makes no functional difference.
+        matches = [dims for cm_, dims in self._cell_measures_and_dims if
+                   cm_ is cell_measure]
+
+        if not matches:
```

Comment:
@marqh Your change is confused with the `coord_dims` method ...

This should should be ...

```python
if not matches:
    raise iris.exceptions.CellMethodNotFoundError(cell_measure.name())
```